src/General/Settings.coffee

```diff
@@ -693,6 +716,31 @@ Settings =
       Conf['selectedArchives'] = selectedArchives
       Redirect.selectArchives()
 
+  captchaServiceDomain: ->
+    $.get 'captchaServiceKey', Conf['captchaServiceKey'], ({captchaServiceKey}) =>
+      keyInput = $('[name=captchaServiceKey]')
+      keyInput.value = captchaServiceKey[@value.trim()] or ''
+      keyInput.disabled = !@value.trim()
+
+  captchaServiceKey: ->
+    domain = Conf['captchaServiceDomain']
+    value = @value.trim()
+    Conf['captchaServiceKey'][domain] = value
+    $.get 'captchaServiceKey', Conf['captchaServiceKey'], ({captchaServiceKey}) ->
+      captchaServiceKey[domain] = value
+      delete captchaServiceKey[domain] unless value or (domain of Config['captchaServiceKey'][0])
+      Conf['captchaServiceKey'] = captchaServiceKey
+      $.set 'captchaServiceKey', captchaServiceKey
+      Settings.captchaServiceDomainList()
+
+  captchaServiceDomainList: ->
+    list = $.id 'list-captchaServiceDomain'
+    $.rmAll list
+    for domain of Conf['captchaServiceKey']
+      $.add list, $.el 'option',
+        textContent: domain
+    return
+
```

Comment:
For future reference, when you merge stuff in from the master branch, it would be better to do it in a separate merge commit to make it easier to see what you added versus what you just merged in.

Reply:
Sorry, I'm still trying to wrap my head around this. I used the browser to fork and edit changes.
When you say 'separate merge commit', does that mean I should've also included the latest changes to master?

Reply:
I don't know whether it's possible in the browser. On the command line, it would look something like this:

git remote add ccd0 https://github.com/ccd0/4chan-x.git
git fetch ccd0
git merge ccd0/master

This would automatically merge in all the changes I'd made on the master branch since you forked, possibly prompting you to manually resolve conflicts. It would then make a commit with two parents, one being your previous commit, and the other being the latest commit on my master branch.

Reply:
Okay thanks, I'll try to make sure that I attempt to do this in the future. My apologies!